### PR TITLE
refactor: remove repetitive safe_query

### LINKED ISSUE #7001: Remove repetitive safe_query
You'll find a lot of unnecessary and repetitive `safe_query` calls like `safe_query(db, Session, 'id', view_kwargs['session_id'], 'session_id')`

First of all, all of this should be abstracted, manually calling this is the ultimate source of bugs. However, that will take massive restructuring of the project and hence, for now, just refactor it to be simpler, like `safe_query_kwargs(Session, view_kwargs, 'session_id')`, that's all the info it needs to do its thing, no need for extra verbosity. If possible, change to `safe_query_by_id(Session, view_kwargs['session_id'])`

### PR DESCRIPTION
**Pull request by satya7289**
Fixes #7001 

#### Short description of what this resolves:
change the all safe_query(db, Session, 'id', view_kwargs['session_id'], 'session_id') into safe_query_kwargs(Session, view_kwargs, 'session_id')

#### Changes proposed in this pull request:

-  Remove repetitive safe_query for session

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

File: app/api/microlocations.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 from flask_rest_jsonapi import ResourceDetail, ResourceList, ResourceRelationship
 
 from app.api.bootstrap import api
-from app.api.helpers.db import safe_query
+from app.api.helpers.db import safe_query, safe_query_kwargs
```
Comment (niranjan94) [inline app/api/microlocations.py:4]: Codacy Issue found: [Unused safe_query imported from app.api.helpers.db](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536746)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536746&bid=18150255)

File: app/api/helpers/db.py [hunk 2]
```diff
@@ -129,3 +129,21 @@ def get_new_slug(model, name):
...
+    :return:
+    """
+    return safe_query_without_soft_deleted_entries(
+        model,
+        'id',
+        kwargs[parameter_name],
+        parameter_name,
+        # TODO(Areeb): Check that only admin can pass this parameter
+        request.args.get('get_trashed') != 'true',
+    )
```
Comment (niranjan94) [inline app/api/helpers/db.py:149]: Codacy Issue found: [Final newline missing](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536746)

File: app/api/tracks.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 from flask_rest_jsonapi import ResourceDetail, ResourceList, ResourceRelationship
 
 from app.api.bootstrap import api
-from app.api.helpers.db import safe_query
+from app.api.helpers.db import safe_query, safe_query_kwargs
```
Comment (niranjan94) [inline app/api/tracks.py:4]: Codacy Issue found: [Unused safe_query imported from app.api.helpers.db](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536746)

File: app/api/session_types.py [hunk 4]
```diff
@@ -1,7 +1,7 @@
 from flask_rest_jsonapi import ResourceDetail, ResourceList, ResourceRelationship
 
 from app.api.bootstrap import api
-from app.api.helpers.db import safe_query
+from app.api.helpers.db import safe_query, safe_query_kwargs
```
Comment (niranjan94) [inline app/api/session_types.py:4]: Codacy Issue found: [Unused safe_query imported from app.api.helpers.db](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536746)

File: app/api/helpers/db.py [hunk 5]
```diff
@@ -129,3 +129,21 @@ def get_new_slug(model, name):
         return slug
     else:
         return '{}-{}'.format(slug, uuid.uuid4().hex)
+
+def safe_query_kwargs(model, kwargs, parameter_name):
```
Comment (iamareebjamal) [inline app/api/helpers/db.py:133]: Wrong place to add the method

**Comment by iamareebjamal** [comment]
Fix hound and codacy issues

File: app/api/helpers/db.py (hunk 5 above)
Comment (satya7289) [reply app/api/helpers/db.py:133]: Where should I place this method?

Comment (iamareebjamal) [reply app/api/helpers/db.py:134]: Near others

File: app/api/helpers/db.py [hunk 6]
```diff
@@ -82,6 +82,25 @@ def safe_query(model, column_name, value, parameter_name):
...
+    """
+    :param model: db Model to be queried
+    :param kwargs: it contains parameter_name's value eg kwargs['event_id']
+                where parameter_name='event_id'
+    :param parameter_name: Name of parameter to be printed in json-api error
+                message eg 'event_id'
+    :return:
+    """
+    return safe_query_without_soft_deleted_entries(
```
Comment (iamareebjamal) [inline app/api/helpers/db.py:94]: You are replacing `safe_query` method, so you should use it instead, not `safe_query_without_soft_deleted_entries`

And move it above ``safe_query_without_soft_deleted_entries

**Comment by iamareebjamal** [comment]
Great job, but a lot of safe_query calls are still remaining

**Comment by satya7289** [comment]
> Great job, but a lot of safe_query calls are still remaining

Yes, do I cover all safe_query calls in this PR?

**Comment by iamareebjamal** [comment]
Yes, obviously the issue will only be resolved once all such calls are handled

**Comment by satya7289** [comment]
> Yes, obviously the issue will only be resolved once all such calls are handled

Ok

**Comment by iamareebjamal** [comment]
Also, calls like this are very similar:

```python
safe_query(EventInvoice, 'identifier', view_kwargs['event_invoice_identifier'], 'event_invoice_identifier')
```

The only difference is `identifier` instead of `id`, so `safe_query_kwargs` can take a keyword argument, `column` with default value `id`, and then the call will change to `safe_query(EventInvoice, view_kwargs, 'event_invoice_identifier', column='identifier')`

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ app/api/feedbacks.py  -1
+ app/api/custom_forms.py  -1
+ app/api/event_copyright.py  -1
+ app/api/tax.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536746&bid=18150255)

**Comment by satya7289** [comment]
> Also, calls like this are very similar:
> 
> ```python
> safe_query(EventInvoice, 'identifier', view_kwargs['event_invoice_identifier'], 'event_invoice_identifier')
> ```
> 
> The only difference is `identifier` instead of `id`, so `safe_query_kwargs` can take a keyword argument, `column` with default value `id`, and then the call will change to `safe_query(EventInvoice, view_kwargs, 'event_invoice_identifier', column='identifier')`

ok

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ app/api/custom_forms.py  -2
+ app/api/feedbacks.py  -2
+ app/api/tickets.py  -1
+ app/api/tax.py  -2
+ app/api/event_copyright.py  -1
+ app/api/event_types.py  -1
+ app/api/event_sub_topics.py  -1
+ app/api/event_topics.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536746&bid=18150255)

File: app/api/orders.py [hunk 7]
```diff
@@ -631,7 +632,7 @@ def before_get(self, args, kwargs):
             )
             kwargs['id'] = order.id
         elif kwargs.get('id'):
-            order = safe_query(Order, 'id', kwargs['id'], 'id')
+            order = safe_query_kwargs(Order, kwargs, 'id')
```
Comment (iamareebjamal) [inline app/api/orders.py:635]: Use `safe_query_by_id` here

**Comment by iamareebjamal** [comment]
Great job. Final thing, please change the PR title to match semantic guidelines, see other merged PRs for reference

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ app/api/event_topics.py  -1
+ app/api/event_copyright.py  -1
+ app/api/feedbacks.py  -2
+ app/api/custom_forms.py  -2
+ app/api/event_types.py  -1
+ app/api/tax.py  -2
+ app/api/tickets.py  -1
+ app/api/event_sub_topics.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536746&bid=18150255)

**Comment by satya7289** [comment]
> Great Job!

Thank you so much.

**Pull request merged**